src/main/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImpl.java

```diff
@@ -184,6 +193,39 @@ public String getActualBeanstalkAmazonLinuxPlatform() {
 		}
 		return actualVersion;
 	}
+	
+	public String getLatestImageIdForImagePipelineArn(String imagePipelineArn) {
+		if(imagePipelineArn == null) {
+			return null;
+		}
+		String latestDate = null;
+		String latestImage = null;
+		String nextPageToken = null;
+		while (true) {
+			ListImagePipelineImagesRequest listImagePipelineImagesRequest = new ListImagePipelineImagesRequest().
+					withImagePipelineArn(imagePipelineArn).
+					withNextToken(nextPageToken);
+			ListImagePipelineImagesResult result = imageBuilder.listImagePipelineImages(listImagePipelineImagesRequest);
+
+			for (ImageSummary imageSummary : result.getImageSummaryList()) {
+				if (!IMAGE_AVAILABLE_STATE.equalsIgnoreCase(imageSummary.getState().getStatus())) {
```

Comment:
There might be a Status enum in the sdk.

Reply:
Maybe add some logging around the selected/skipepd image?

Reply:
Done!